README.md

```diff
@@ -44,6 +44,299 @@ This is an experimental framework that aims to combine the lessons learned from
+targets by name from within the annotation, see below for an example.
+
+</details>
+
+#### Annotations
+
+There are several predefined **annotation types** in `pytorch_ie.annotations`, however, feel free to define your own.
+Annotations have to be dataclasses that subclass `pytorch_ie.core.Annotation`. They also need to be hashable and
+immutable. The following is a simple example:
+
+```python
+@dataclass(eq=True, frozen=True)
+class SimpleLabeledSpan(Annotation):
+    start: int
+    end: int
+    label: str
+```
+
+<details>
+<summary>
+
+##### Accessing Target Content
+
+</summary>
+
+We can expand the above example a little to have a nice string representation:
+
+```python
+@dataclass(eq=True, frozen=True)
+class LabeledSpan(Annotation):
+    start: int
+    end: int
+    label: str
+
+    def __str__(self) -> str:
+        if self.targets is None:
+            return ""
+        return str(self.target[self.start : self.end])
+```
+
+The content of `self.target` is lazily assigned as soon as the annotation is added to a document.
+
+Note that this now expects a single `collections.abc.Sequence` as `target`, e.g.:
+
+```python
+my_spans: AnnotationList[Span] = annotation_field(target="<NAME_OF_THE_SEQUENCE_FIELD>")
+```
+
+If we have multiple targets, we need to define target names to access them. For this, we need to set the special
+field `TARGET_NAMES`:
+
+```python
+@dataclass(eq=True, frozen=True)
+class Alignment(Annotation):
+    TARGET_NAMES = ("text1", "text2")
+    start1: int
+    end1: int
+    start2: int
+    end2: int
+
+    def __str__(self) -> str:
+        if self.targets is None:
+            return ""
+        # we can access the `named_targets` which has the keys defined in `TARGET_NAMES`
+        span1 = self.named_targets["text1"][self.start1 : self.end1]
+        span2 = self.named_targets["text2"][self.start2 : self.end2]
+        return f'span1="{span1}" is aligned with span2="{span2}"'
+```
+
+This requires to define the annotation container as follows:
+
+```python
+class MyDocumentWithAlignment(Document):
+    text_a: str
+    text_b: str
+    # `named_targets` defines the mapping from `TARGET_NAMES` to data fields
+    my_alignments: AnnotationList[Alignment] = annotation_field(named_targets={"text1": "text_a", "text2": "text_b"})
+```
+
+Note that `text1` and `text2` can also target the same field.
+
+</details>
+<details>
+<summary>
+
+##### (De-)Serialization of Annotations
+
+</summary>
+
+As usual for dataclasses, annotations can be converted to json like objects with `.asdict()`. However, they can be
+also created with `MyAnnotation.fromdict(dct, annotation_store)`. Both methods are required because documents and
+their annotations are created on the fly when working with PIE datasets (see below).
+
+Sometimes, it is required to overwrite both methods. This is the case when targeting another annotation field. Consider
```

Comment:
this example confuses me. up to now, all annotations had basic types like str, int. And for Document class above, you state that annotations have to be inside this AnnotationList container, otherwise it doesn't work. Now, in lines 202-203, you introduce "annotations" with "targets" that use a totally different 'syntax' ? (i.e. also created without the 'annotation_field(target='entities')' ? 
(I think maybe the use of the word 'targeting' in l196 is confusing. Here, you're using the annotation type 'Span' as the class of the BinaryRelation's member fields head and tail, but not as the 'target' annotation field in the sense you used earlier, correct?

Reply:
I think this needs to be introduced a bit more clearly, and outside of the scope of the serialization

Reply:
yes, good idea